packages/blade/src/components/Card/CardHeader.tsx

```diff
@@ -0,0 +1,194 @@
+  useInsideCard('CardHeaderText');
+
+  return <Text {...props} />;
+};
+CardHeaderText.componentId = Components.CardHeaderText;
+
+const CardHeaderLink = (props: LinkProps): React.ReactElement => {
+  useInsideCard('CardHeaderLink');
+
+  return <Link {...props} />;
+};
+CardHeaderLink.componentId = Components.CardHeaderLink;
+
+type CardHeaderIconButtonProps = Omit<
+  ButtonProps,
+  'variant' | 'size' | 'iconPosition' | 'isFullWidth' | 'children'
+> & {
+  icon: IconComponent;
+};
+
+const CardHeaderIconButton = (props: CardHeaderIconButtonProps): React.ReactElement => {
+  useInsideCard('CardHeaderIconButton');
+
+  return (
+    <Box width={makeSpace(minHeight.xsmall)}>
+      <Button {...props} variant="tertiary" size="xsmall" iconPosition="left" isFullWidth />
+    </Box>
+  );
+};
+CardHeaderIconButton.componentId = Components.CardHeaderIconButton;
+
+const getComponentId = (comp: React.ReactNode): string | null => {
+  if (!React.isValidElement(comp)) return null;
+  return (comp.type as any)?.componentId;
+};
+
+const isOfComponentId = (comp: React.ReactNode, id: string): boolean => {
+  return getComponentId(comp) === id;
+};
+
+const CardHeader = ({ children }: CardHeaderProps): React.ReactElement => {
+  useInsideCard('CardHeader');
+
+  return (
+    <Box marginBottom="spacing.7">
+      <Box
+        marginBottom="spacing.7"
+        display="flex"
+        flexDirection="row"
+        justifyContent="space-between"
+      >
+        {children}
+      </Box>
+      <Divider />
+    </Box>
+  );
+};
+
+type CardHeaderLeadingProps = {
+  title: string;
+  subtitle?: string;
+  prefix?: React.ReactNode;
+  suffix?: React.ReactNode;
+};
+const CardHeaderLeading = ({
+  title,
+  subtitle,
+  prefix,
+  suffix,
+}: CardHeaderLeadingProps): React.ReactElement => {
+  useInsideCard('CardHeaderLeading');
+
+  if (prefix && !isOfComponentId(prefix, Components.CardHeaderIcon)) {
+    throw new Error(
+      `[Blade CardHeaderLeading]: Only \`${Components.CardHeaderIcon}\` component is accepted in prefix`,
+    );
+  }
+
+  if (suffix && !isOfComponentId(suffix, Components.CardHeaderCounter)) {
+    throw new Error(
+      `[Blade CardHeaderLeading]: Only \`${Components.CardHeaderCounter}\` component is accepted in prefix`,
+    );
+  }
+
+  return (
+    <Box flex={1} display="flex" flexDirection="row">
+      <Box marginRight="spacing.3" alignSelf="center" display="flex">
+        {prefix}
+      </Box>
+      <Box>
+        <Box display="flex" flexDirection="row" alignItems="center" flexWrap="wrap">
+          <Heading size="small" variant="regular" type="normal">
+            {title}
+          </Heading>
+          <Box marginLeft="spacing.3">{suffix}</Box>
+        </Box>
+        <Text variant="body" size="small" weight="regular">
+          {subtitle}
+        </Text>
+      </Box>
+    </Box>
+  );
+};
+
+type CardHeaderTrailingProps = {
+  visual?: React.ReactNode;
+};
+
+const headerTrailingAllowedComponents = [
```

Comment:
This array holds the `componentId (string)` of the allowed components, not the components itself. Then we check it via `getComponentId()`